File: homework-10/src/main/ui/src/service/BookService.js
import axios from 'axios';

class BookService{

    getBookById(id){
        return axios.get("/books/" + id);
    }

    getBooks(params){
        return axios.get("/books", {params: params});
    }

    createBook(book){
        return axios.post("/books/", book);
    }

    updateBook(book){
        return axios.put("/books/" + book.id, {title: book.title, authorId: book.authorId, genresId: book.genresId});
    }

    deleteBook(id){
        return axios.delete("/books/" + id);
    }

}

export default new BookService();

File: homework-10/src/main/java/org/course/service/interfaces/BookService.java
package org.course.service.interfaces;

import org.course.api.requests.BookListRequest;
import org.course.api.requests.BookRequest;
import org.course.api.responces.BookInfoResponse;
import org.course.api.responces.BookListResponse;
import org.course.domain.Book;

public interface BookService {

    BookListResponse getBooks(BookListRequest request);

    BookListResponse getBooksByGenre(BookListRequest request);

    BookListResponse getBooksByQuery(BookListRequest request);

    BookInfoResponse getBookById(String id);

    Book createBook(BookRequest request);

    Book updateBook(String id, BookRequest request);

    void deleteBook(String id);
}
